### PR TITLE
Validation: wrap local chain info maps with mutexes

### PR DESCRIPTION
**Pull request by vdamle**
* Appears that there's a race when attempting to update local chain ID/Name maps from multiple tests which are kicked off concurrently.
* Also refactored the code a bit to wrap the global vars into a struct

## Checklist
- [ ] ~I have declared the chain at the appropriate [Superchain Level](../docs/glossary.md#superchain-level-and-rollup-stage).~
- [x] I have run `just validate <chain-id>` locally to ensure all local validation checks pass. 
- [x] I have run `just codegen` to ensure that the chainlist and other generated files are up-to-date and include my chain.

ref: https://github.com/ethereum-optimism/superchain-registry/issues/668

### CODE REVIEW

File: validation/uniqueness_test.go [hunk 1]
```diff
@@ -20,23 +21,39 @@ type uniqueProperties struct {
...
+		return fmt.Errorf("Chain with ID %d is duplicated", chainId)
+	}
+	c.chainIDMutex.RUnlock()
 
-func (s chainNameSet) AddIfUnique(name string) error {
-	if s[name] {
-		return fmt.Errorf("Chain with name %s is duplicated", name)
+	// Check if chainName is unique
+	c.chainNameMutex.RLock()
+	if c.localChainIds[chainId] {
```
Comment (geoknee) [inline validation/uniqueness_test.go:47]: Don't you want to check `c.localChainNames` here?

Also, where do you write to `c.localChainIds`?

**Review by geoknee — COMMENTED** [review]
Did you consider using https://pkg.go.dev/sync#Map ?

File: validation/uniqueness_test.go (hunk 1 above)
Comment (vdamle) [reply validation/uniqueness_test.go]: This was a copy-paste error. moved to `sync.Map` per your other suggestion, this is no longer relevant.

**Comment by vdamle** [comment]
> Did you consider using https://pkg.go.dev/sync#Map ?

Thanks for the suggestion, agree that's much simpler to use for this case and modified code accordingly.

**Review by geoknee — APPROVED** [review]

**Pull request merged**